### PR TITLE
Libvirt test create network and storage

### PR DESCRIPTION
**Pull request by bookinabox**
Relies on: #1244 

Creates temporary network and storage if the provided libvirt_network and/or libvirt_storage are not networks/storage pools that are currently enabled.

Destroys network and storage on DeleteVPC (which is triggered when provisioning and teardown are both enabled)

### CODE REVIEW

**Comment by bookinabox** [comment]
I am thinking of changing how the storage works. The pool type is a directory pool, so I created a directory in /tmp. While the pool itself is destroyed, the directory is not.

**Comment by bookinabox** [comment]
I do not know of a way to cleanly delete the directory of a transient storage pool. I'm thinking I can either change to a persistent storage pool, which I then define, start, use, destroy, delete, undefine. Or I can parse the storage xml description and use os commands to delete the directory. I'm leaning towards the second option, but I'm sure there are aspects I am not taking into account.

**Comment by bookinabox** [comment]
Changed it to delete the volumes and just ignore the empty folder in /tmp, but if there are any opinions on if the other ways are better, I would be happy to change it.

**Comment by bookinabox** [comment]
Changed the xml to libvirtxml.

File: test/provisioner/provision_libvirt.go [hunk 1]
```diff
@@ -135,11 +134,99 @@ func (l *LibvirtProvisioner) CreateVPC(ctx context.Context, cfg *envconf.Config)
 	)
 
 	if _, err := l.conn.LookupNetworkByName(l.network); err != nil {
-		return fmt.Errorf("Network '%s' not found. It should be created beforehand", l.network)
+		// Check if network is defined, but not enabled
+		defined_nets, err := l.conn.ListDefinedNetworks()
```
Comment (wainersm) [inline test/provisioner/provision_libvirt.go:136]: Please use camel-case with Go: s/defined_nets/definedNets/

**Comment by wainersm** [comment]
@bookinabox hi! I'm getting an error:

```
[wmoschet@wainer-laptop cloud-api-adaptor]$ ./run_libvirt_e2e.sh 
go test -v -tags=libvirt -timeout 40m -count=1 ./test/e2e
time="2023-08-14T12:32:59-03:00" level=info msg="Do setup"
time="2023-08-14T12:32:59-03:00" level=info msg="Cluster provisioning"
time="2023-08-14T12:32:59-03:00" level=info msg="Network peer-pods-net is not defined. Creating a new temporary network"
time="2023-08-14T12:32:59-03:00" level=info msg="Using 192.168.123.1 as the network ip for peer-pods-net"
time="2023-08-14T12:32:59-03:00" level=info msg="Storage peer-pods-storage is not defined. Creating a new temporary storage pool"
time="2023-08-14T12:32:59-03:00" level=info msg="Created temporary Pool peer-pods-storage"
You need to define api_ip in your parameters file
F0814 12:32:59.595380  315187 env.go:369] Setup failure: exit status 1
FAIL    github.com/confidential-containers/cloud-api-adaptor/test/e2e   0.451s
FAIL
make: *** [Makefile:91: test-e2e] Error 1
```

**Comment by bookinabox** [comment]
> @bookinabox hi! I'm getting an error:
> 
> ```
> [wmoschet@wainer-laptop cloud-api-adaptor]$ ./run_libvirt_e2e.sh 
> go test -v -tags=libvirt -timeout 40m -count=1 ./test/e2e
> time="2023-08-14T12:32:59-03:00" level=info msg="Do setup"
> time="2023-08-14T12:32:59-03:00" level=info msg="Cluster provisioning"
> time="2023-08-14T12:32:59-03:00" level=info msg="Network peer-pods-net is not defined. Creating a new temporary network"
> time="2023-08-14T12:32:59-03:00" level=info msg="Using 192.168.123.1 as the network ip for peer-pods-net"
> time="2023-08-14T12:32:59-03:00" level=info msg="Storage peer-pods-storage is not defined. Creating a new temporary storage pool"
> time="2023-08-14T12:32:59-03:00" level=info msg="Created temporary Pool peer-pods-storage"
> You need to define api_ip in your parameters file
> F0814 12:32:59.595380  315187 env.go:369] Setup failure: exit status 1
> FAIL    github.com/confidential-containers/cloud-api-adaptor/test/e2e   0.451s
> FAIL
> make: *** [Makefile:91: test-e2e] Error 1
> ```

@wainersm It relies on #1244 to get the ip. Should I rebase and merge these into one PR then?

**Pull request closed without merging**